src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.ts

```diff
@@ -236,8 +301,17 @@ export const AzureComputeModalBase = ({
         viewMode === 'deleteEnvironment',
         () =>
           Utils.cond(
-            [doesRuntimeExist(), () => Ajax().Runtimes.runtimeV2(workspaceId, currentRuntime.runtimeName).delete(deleteDiskSelected)], // delete runtime
-            [!!persistentDiskExists, () => leoDiskProvider.delete(currentPersistentDiskDetails)] // delete disk
+            [
+              doesRuntimeExist(),
+              () =>
+                currentRuntime
+                  ? Ajax().Runtimes.runtimeV2(workspaceId, currentRuntime.runtimeName).delete(deleteDiskSelected)
+                  : {},
+            ], // delete runtime
+            [
+              !!persistentDiskExists,
+              () => (currentPersistentDiskDetails ? leoDiskProvider.delete(currentPersistentDiskDetails) : {}),
```

Comment:
hmm... this seems to be a small bug. Perhaps change the condition to check `currentPersistentDiskDetails` and then make the call if it exists rather than having a ternary nested in a `Utils.cond` ?

Reply:
🤔 maybe I can show you at parking lot, but the linter does not seem to acknowledge the condition and complains that the `currentPersistentDiskDetails` can be undefined, same for the condition right above too

Reply:
@jdcanas The `cond` from the logic utils function seems to be behaving similarly to the `Utils.cond` and do not play well with the linter unless the condition check is nested inside. I think it should be good to merge as is, but lmk if you think otherwise